Makefile

```diff
@@ -7,19 +7,23 @@ help:
 	@echo "clean				Remove artifacts and standardize repo."
 
 # CORE ]------------------------------------------------------------------------
-test:
-	black --check .
+test: deps
+	. venv/bin/activate ;\
+	black --check . --exclude venv
 
 config:
 	touch spotitude.config && \
 	printf "[DEFAULT]\nUSERNAME=\nSCOPE=user-top-read playlist-modify-private\nREDIRECT_URI=http://localhost:8080\nCLIENT_ID=\nCLIENT_SECRET=" > spotitude.config
 
-deps:
-	pip3 install -r requirements.txt
+deps: requirements.txt
+	test -d venv || (python3 -m venv venv && \
+	venv/bin/pip3 install -r requirements.txt && \
+	touch venv/bin/activate)
 
 server:
 	python3 -m http.server 8080
 
-clean:
-	black . && \
-	rm -rf *.html *.csv .cache-* __pycache__
+clean: deps
+	. venv/bin/activate ;\
```

Comment:
What would be the purpose to activate the environment in `clean`?

Reply:
This is done to ensure we pick black from the virtual environment (as it's in the requirement)